webclient/pid/client.py

```diff
@@ -0,0 +1,107 @@
+import logging
+
+import requests
+from django.conf import settings
+from django.utils.translation import gettext as _
+from dokument.models import Dokument
+from lokalita.models import Lokalita
+from pas.models import SamostatnyNalez
+from pid.model_serializers import DokumentSerializer, LokalitaSerializer, SamostatnyNalezSerializer
+from requests.auth import HTTPBasicAuth
+
+from webclient.settings.base import DATACITE_URL, DATACITE_USER, DATACITE_USER_PASSWORD
+
+logger = logging.getLogger(__name__)
+
+
+class DoiWriteError(Exception):
+    pass
+
+
+class DoiHasBeenAlreadyPublishedError(DoiWriteError):
+    pass
+
+
+class DoiHasNotBeenPublishedYetError(DoiWriteError):
+    pass
+
+
+class DoiNoTransactionError(DoiWriteError):
+    pass
+
+
+class DigitalObjectIdentifierClient:
+    auth = HTTPBasicAuth(DATACITE_USER, DATACITE_USER_PASSWORD)
+    headers = {"Content-Type": "application/vnd.api+json"}
+
+    def __init__(self, record):
+        self.record = record
+        if isinstance(self.record, Dokument):
```

Comment:
Nebylo by lepsi mit mapu class:serializer nez else if?

Reply:
Upravil jsem to.